torch_geometric/explain/explanation.py

```diff
@@ -129,3 +131,188 @@ def _apply_masks(
+                    status = False
+                    warn_or_raise(
+                        f"Expected a 'node_feat_mask' of shape "
+                        f"{list(x_dict[key].size())} (got shape "
+                        f"{list(self.node_feat_mask[key].size())})",
+                        raise_on_error)
+
+        if 'edge_feat_mask' in self:
+            for key in edge_attr_dict.keys():
+                if key not in self.edge_feat_mask:
+                    status = False
+                    warn_or_raise(
+                        f"Expected an 'edge_feat_mask' for edge type {key}, "
+                        f"but got none", raise_on_error)
+                elif edge_attr_dict[key].size(
+                ) != self.edge_feat_mask[key].size():
+                    status = False
+                    warn_or_raise(
+                        f"Expected a 'edge_feat_mask' of shape "
+                        f"{list(edge_attr_dict[key].size())} (got shape "
+                        f"{list(self.edge_feat_mask[key].size())})",
+                        raise_on_error)
+
+        return status
+
+    def get_explanation_subgraph(self) -> 'HeteroExplanation':
+        r"""Returns the induced subgraph, in which all nodes and edges with
+        zero attribution are masked out."""
+        return self._apply_masks(
+            node_mask={
+                key: value > 0
+                for key, value in self.node_mask.items()
+            } if 'node_mask' in self else None,
+            edge_mask={
+                key: value > 0
+                for key, value in self.edge_mask.items()
+            } if 'edge_mask' in self else None,
+        )
+
+    def get_complement_subgraph(self) -> 'HeteroExplanation':
+        r"""Returns the induced subgraph, in which all nodes and edges with any
+        attribution are masked out."""
+        return self._apply_masks(
+            node_mask={
+                key: value == 0
+                for key, value in self.node_mask.items()
+            } if 'node_mask' in self else None,
+            edge_mask={
+                key: value == 0
+                for key, value in self.edge_mask.items()
+            } if 'edge_mask' in self else None,
+        )
+
+    def _apply_masks(
+        self,
+        node_mask: Optional[Tensor] = None,
+        edge_mask: Optional[Tensor] = None,
+    ) -> 'HeteroExplanation':
+        out = copy.copy(self)
+
+        edge_index_dict = None
+        edge_attr_dict = None
+        for key, value in out.node_items():
+            if key == "edge_index_dict":
+                edge_index_dict = value
+            elif key == "edge_attr_dict":
+                edge_attr_dict = value
+
+        if edge_mask is not None:
+            for key, value in self.edge_mask.items():
+                edge_index_dict[key] = edge_index_dict[key][:, value]
+                edge_attr_dict[key] = edge_attr_dict[key][value]
```

Comment:
```suggestion
            for edge_type in self.edge_types:
               out[edge_type].edge_index = out[edge_type].edge_index[:, value]
```
This might be cleaner. Also we need to check if `edge_attr` is in `out[edge_type]` before masking it.